demos/python_demos/object_detection_demo_yolov3_async/object_detection_demo_yolov3_async.py

```diff
@@ -89,9 +101,29 @@ def __init__(self, param, side):
 
             self.isYoloV3 = True # Weak way to determine but the only one.
 
-    def log_params(self):
-        params_to_print = {'classes': self.classes, 'num': self.num, 'coords': self.coords, 'anchors': self.anchors}
-        [log.info("         {:8}: {}".format(param_name, param)) for param_name, param in params_to_print.items()]
+
+class Modes(Enum):
+    USER_SPECIFIED = 0
+    MIN_LATENCY = 1
+
+
+class Mode():
+    def __init__(self, value):
+        self.current = value
+
+    def next(self):
+        if self.current.value + 1 < len(Modes):
+            self.current = Modes(self.current.value + 1)
+        else:
+            self.current = Modes(0)
```

Comment:
Looks like this entire class could be replaced with `itertools.cycle(Modes)`.

Reply:
@IRDonch,
The iterator returned by itertools.cycle(Modes) does not allow me to get the "current" value, only the "next" one. I need the "current" value most of the times.

Of course, I can create a variable for storing the current value, but in that case I will also need to wrap it into dict or something in order to be able to pass it into callback function by reference. Or I can make the variable global, which is worse.
So, I'm not sure that replacing this class with itertools.cycle will result in better code.

Reply:
OK, fair enough.